lib/rucio/core/distance.py

```diff
@@ -141,3 +142,28 @@ def update_distances(src_rse_id=None, dest_rse_id=None, parameters=None, session
         query.update(params)
     except IntegrityError as error:
         raise exception.RucioException(error.args)
+
+
+@read_session
+def export_distances(session=None):
+    """
+    Get distances between all the rses.
+
+    :param session: The database session to use.
+
+    :returns distance: dictionary of dictionaries with all the distances.
+    """
+
+    distances = {}
+    try:
+        query = session.query(Distance)
+        for distance in query.all():
+            src_name = get_rse_name(distance.src_rse_id)
+            dst_name = get_rse_name(distance.dest_rse_id)
+            key = src_name + '__' + dst_name
+            distances[key] = {}
+            for k, v in distance:
+                distances[key][k] = v
```

Comment:
a join with the rse table is more efficient.

Reply:
This is not to get the distance between two sites, but all the distances. I need to do a full table scan anyways.

Reply:
Aaaaaah... I see.. I will fix it. Thanks for the clarification!